lib/auth/init.go

```diff
@@ -1374,3 +1360,30 @@ func migrateMFADevices(ctx context.Context, asrv *Server) error {
 
 	return nil
 }
+
+// DELETE IN: 8.0.0
+// migrateClusterID moves the cluster ID information
+// from ClusterConfig to ClusterName.
+func migrateClusterID(ctx context.Context, asrv *Server) error {
+	clusterConfig, err := asrv.ClusterConfiguration.GetClusterConfig()
+	if err != nil {
+		if trace.IsNotFound(err) {
+			return nil
+		}
+		return trace.Wrap(err)
+	}
+
+	clusterName, err := asrv.ClusterConfiguration.GetClusterName()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	log.Debugf("Migrating cluster ID %q from ClusterConfig to ClusterName.", clusterConfig.GetLegacyClusterID())
+
+	clusterName.SetClusterID(clusterConfig.GetLegacyClusterID())
+	if err := asrv.ClusterConfiguration.UpsertClusterName(clusterName); err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
nit: only do this `if clusterName.GetClusterID() == ""`
users should not see a migration log on every start.